DOCS-8334 Unified Service Tagging Update (#23951)
DataDog/documentation: content/en/getting_started/tagging/unified_service_tagging.md

Author: I've added a reference in the Hostname and Tagging doc page—let me know if that makes sense!

Reviewer: Let me loop in our PM to confirm

- link: "/tracing/version_tracking"
  tag: "Documentation"
  text: "Use Version tags within Datadog APM to monitor deployments"
- link: "https://www.datadoghq.com/blog/unified-service-tagging/"
  tag: "Blog"

>   text: "Tags: set once, access everywhere"

Reviewer 2: ```suggestion
  text: "Tags: Set once, access everywhere"
```

### Serverless environment

For more information about AWS Lambda functions, see [how to connect your Lambda telemetry using tags][15].

### OpenTelemetry environment

In OpenTelemetry environments, you can use the [OpenTelemetry Collector][16] to collect and batch telemetry data (like metrics, traces, and logs) by leveraging the [Datadog Exporter][17] to send the data to Datadog without the Datadog Agent.

> Optionally, you can send telemetry data from [applications instrumented with the OpenTelemetry SDKs with the Datadog Agent][20], which collects the data for Datadog.

Reviewer: Our stance is to use one or the other, not both

To set up unified service tagging in an OpenTelemetry environment:

> 1. Depending on the OpenTelemetry component you are using, set the following attributes to ensure proper tagging:

Reviewer 2: The "following attributes" wording feels a little confusing, since an explicit list of attributes does not follow, just two examples.

Maybe you could say something like "To ensure proper tagging, set the required attributes for the OpenTelemetry component you are using, as shown in the examples below." Then the examples could just say "With the Python OpenTelemetry SDK:"

Author: ```suggestion
1. Set the required attributes to ensure proper tagging for the OpenTelemetry component you are using.
```

>    For example, with the Python OpenTelemetry SDK:

Reviewer: I would have a tab per example

   ```python
   from opentelemetry.sdk.resources import Resource
   from opentelemetry.sdk.trace import TracerProvider

   resource = Resource(attributes={
      "service.name": "<SERVICE>",
      "deployment.environment": "<ENV>",
      "service.version": "<VERSION>"
    })
    tracer_provider = TracerProvider(resource=resource)
    ```

    For example, with the OpenTelemetry Collector:

    ```yaml
    receivers:
    otlp:
      protocols:
        grpc:
        http:

    processors:
      batch:

    exporters:
      datadog:
        api:
          key: "<DATADOG_API_KEY>"
        metrics:
          attributes:
            service.name: "<SERVICE>"
            deployment.environment: "<ENV>"

>             service.version: "<VERSION>"

Reviewer: There is no such `metrics::attributes` key, you would need to use the [transform processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/transformprocessor) for this. I don't think we need a full Collector configuration here, just the transform processor configuraiton

    service:
      pipelines:
        traces:
          receivers: [otlp]
          processors: [batch]
          exporters: [datadog]
    ```

   

> 1. To [correlate OpenTelemetry language SDK logs and traces][18], translate the OpenTelemetry `TraceId` and `SpanId` properties (a 128-bit unsigned integer and 64-bit unsigned integer represented as a 32-character and 16-character hexadecimal string) into the Datadog format and add them as attributes in your logs. 

Reviewer: We have a dedicated page for this, right? I don't think we need to repeat it here

> 1. To [correlate OpenTelemetry language SDK logs and traces][18], translate the OpenTelemetry `TraceId` and `SpanId` properties (a 128-bit unsigned int and 64-bit unsigned int represented as a 32-hex-character and 16-hex-character lowercase string) into the Datadog format and add them as attributes in your logs. 

Author: ```suggestion
1. To [correlate OpenTelemetry language SDK logs and traces][18], translate the OpenTelemetry `TraceId` and `SpanId` properties (a 128-bit unsigned integer and 64-bit unsigned integer represented as a 32-character and 16-character hexadecimal string) into the Datadog format and add them as attributes in your logs. 
```

### Serverless environment

For more information about AWS Lambda functions, see [how to connect your Lambda telemetry using tags][15].

### OpenTelemetry environment

> In OpenTelemetry environments, you can use the [OpenTelemetry Collector][16] to collect and batch telemetry data (like metrics, traces, and logs) and the [OpenTelemetry Exporter][17] to send the data to Datadog (without the Datadog Agent).

Reviewer: We also support using the Datadog Agent as an OpenTelemetry setup https://docs.datadoghq.com/opentelemetry/ we should clarify what to do in that situation

> To setup unified service tagging in an OpenTelemetry environment:

Reviewer 2: ```suggestion
To set up unified service tagging in an OpenTelemetry environment:
```

> 1. Set the following attributes in your OpenTelemetry configuration to ensure proper tagging:

Author: ```suggestion
1. Depending on the OpenTelemetry component you are using, set the following attributes to ensure proper tagging:
```

   ```yaml
   service.name: "<SERVICE>"
   deployment.environment: "<ENV>"
   service.version: "<VERSION>"
   ```
   

> 1. To [correlate OpenTelemetry language SDK logs and traces][18], translate the OpenTelemetry `TraceId` and `SpanId` properties into the Datadog format, and ensure your logs are sent as JSON files. Your language-level logs must become Datadog attributes for the trace-log correlation to work.

Reviewer: > ensure your logs are sent as JSON files

It's unclear to me what you mean by this. There is an [OTLP/JSON format](https://opentelemetry.io/docs/specs/otlp/#json-protobuf-enco), but using this is not required for consuming logs: the OpenTelemetry Collector supports both.

> Your language-level logs must become Datadog attributes

I also don't understand this, what does that mean?

   The following table outlines the key differences between OpenTelemetry and Datadog schemas for tracing and unified service tagging:

   | Domain | OpenTelemetry | Datadog |
   |---|---|---|

>    | Tracing | `TraceId` and `SpanId` are stored in OpenTelemetry as 32-hex character and 16-hex character. | Stored in Datadog as 64-bit unsigned int. |

Reviewer 2: ```suggestion
   | Tracing | `TraceId` and `SpanId` are stored in OpenTelemetry as strings of 32 hex characters and 16 hex characters, respectively. | Stored in Datadog as 64-bit unsigned int. |
```
I'm not 100% sure this was the intended meaning, but the way it was phrased felt odd, since an individual character would just be described as hexadecimal, not 32-hex or 16-hex.

[13]: https://www.chef.io/
[14]: https://www.ansible.com/
[15]: /serverless/configuration/#connect-telemetry-using-tags
[16]: /opentelemetry/collector_exporter/
[17]: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/datadogexporter
[18]: /tracing/other_telemetry/connect_logs_and_traces/

> [19]: /tracing/other_telemetry/connect_logs_and_traces/opentelemetry/

Author: ```suggestion
[19]: /tracing/other_telemetry/connect_logs_and_traces/opentelemetry/
[20]: /opentelemetry/interoperability/otlp_ingest_in_the_agent/
```
